qa/tasks/s3tests.py

```diff
@@ -169,27 +191,30 @@ def create_users(ctx, config):
                         'radosgw-admin',
                         '-n', client_with_id,
                         'caps', 'add',
-                        '--uid', s3tests_conf[section]['user_id'],
+                        '--uid', conf[section]['user_id'],
                         '--caps', 'oidc-provider=*',
                         '--cluster', cluster_name,
                     ],
                 )
 
     if "TOKEN" in os.environ:
-        s3tests_conf.setdefault('webidentity', {})
-        s3tests_conf['webidentity'].setdefault('token',os.environ['TOKEN'])
-        s3tests_conf['webidentity'].setdefault('aud',os.environ['AUD'])
-        s3tests_conf['webidentity'].setdefault('sub',os.environ['SUB'])
-        s3tests_conf['webidentity'].setdefault('azp',os.environ['AZP'])
-        s3tests_conf['webidentity'].setdefault('user_token',os.environ['USER_TOKEN'])
-        s3tests_conf['webidentity'].setdefault('thumbprint',os.environ['THUMBPRINT'])
-        s3tests_conf['webidentity'].setdefault('KC_REALM',os.environ['KC_REALM'])
+        conf.setdefault('webidentity', {})
+        conf['webidentity'].setdefault('token',os.environ['TOKEN'])
+        conf['webidentity'].setdefault('aud',os.environ['AUD'])
+        conf['webidentity'].setdefault('sub',os.environ['SUB'])
+        conf['webidentity'].setdefault('azp',os.environ['AZP'])
+        conf['webidentity'].setdefault('user_token',os.environ['USER_TOKEN'])
+        conf['webidentity'].setdefault('thumbprint',os.environ['THUMBPRINT'])
+        conf['webidentity'].setdefault('KC_REALM',os.environ['KC_REALM'])
 
     try:
         yield
     finally:
-        for client in config['clients']:
+        for client in config.keys():
             for user in users.values():
+                # don't need to delete keystone users
+                if not user in keystone_users:
+                    continue
```

Comment:
> I'm seeing a new error in Yuri's latest testing, which incorporates this PR (and others). The error involves radosgw-admin user create ... and this seems to be the only new PR that might explain this.

@ivancich i think this is the culprit. the failing rgw/sts job runs two s3tests tasks, and the first is supposed to run `user rm` so the second task can recreate it

this code is still wrong on main, but it doesn't fail there after https://github.com/ceph/ceph/pull/52216 combined the two s3tests tasks into one. i created https://tracker.ceph.com/issues/69741 and am testing https://github.com/ceph/ceph/pull/61596 as a fix on main. if that works, i'll cherry-pick the backport here